tests/rados/test_election_strategies.py

```diff
@@ -222,3 +283,382 @@ def check_mon_status(rados_obj, mon_obj):
             return False
         log.info(f"Mon {mon} is in healthy state: {status} | {desc}")
     return True
+
+
+def add_all_mons_to_disallow_list(mon_obj) -> None:
+    """Method to validate last Mon cannot be added to disallowed leaders list
+
+    Steps:
+    - Get list of all Mons from the cluster
+    - Add all Mons to disallowed leaders list
+    - Validate last allowed Mon cannot be added to disallow list
+    - Revert cluster back to original state by removing all Mons from disallow list
+
+    Args:
+        mon_obj: object MonElectionStrategies
+    Returns:
+        Raises Exception when test fails
+    Example:
+
+    """
+    log.info(
+        "\n -------------------------------------------"
+        "\n Starting validation:"
+        "\n `last allowed Mon` cannot be added to disallow list"
+        "\n -------------------------------------------"
+    )
+
+    mons = list(mon_obj.get_mon_quorum().keys())
+    last_allowed_mon = mons.pop()
+
+    for mon in mons:
+        log.info(f"Adding mon {mon} to disallow list")
+        if not mon_obj.set_disallow_mon(mon=mon):
+            raise AssertionError(f"could not add Mon {mon} to disallowed list")
+        log.info(f"Successfully added mon {mon} to disallow list")
+
+    log.info(
+        f"Adding last remaining allowed mon {last_allowed_mon} to disallow list\n"
+        "Proceeding to validate error generation when last allowed mon is added to disallow list"
+    )
+    try:
+        if mon_obj.set_disallow_mon(mon=last_allowed_mon):
```

Comment:
Update set_disallow_mon() call to add `pretty_print=True`, so that error stream gets added in logs

Reply:
Added pretty_print=True for **set_disallow_mon()**

https://github.com/red-hat-storage/cephci/blob/83e34286c4e52f3067a54ba0b8b151b8345e9bde/tests/rados/monitor_configurations.py#L49-L50

Reply:
Added pretty_print=True for  **remove_disallow_mon()**

https://github.com/red-hat-storage/cephci/blob/83e34286c4e52f3067a54ba0b8b151b8345e9bde/tests/rados/monitor_configurations.py#L60-L76